src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -221,11 +218,13 @@ private void AddMetadataFromCustomProviders(IEnumerable<IFunctionProvider> funct
             _logger.ReadingFunctionMetadataFromProvider(MetadataProviderName);
 
             var functionProviderTasks = new List<Task<ImmutableArray<FunctionMetadata>>>();
+            var metadataProviderTimeout = _scriptOptions.Value.MetadataProviderTimeout;
```

Comment:
As long as we go through options setup, this should never be null. But do we want to protect extra just in case it somehow is? I am trying to avoid having to set a default value here as well / avoid duplication of logic